coordinator/cluster/cluster.go

```diff
@@ -38,20 +40,37 @@ func New(config LoadAgentClusterConfig, ltConfig loadtest.Config, log *mlog.Logg
 	if err := defaults.Validate(config); err != nil {
 		return nil, fmt.Errorf("could not validate configuration: %w", err)
 	}
-	agents := make([]*agent.LoadAgent, len(config.Agents))
-	errMap := make(map[*agent.LoadAgent]errorTrack)
+	agents := make([]*client.Agent, len(config.Agents))
+	errMap := make(map[*client.Agent]errorTrack)
 	for i := 0; i < len(agents); i++ {
-		agentConfig := agent.Config{
-			Id:             config.Agents[i].Id,
-			ApiURL:         config.Agents[i].ApiURL,
-			LoadTestConfig: ltConfig,
-		}
-		agent, err := agent.New(agentConfig, log)
+		agent, err := client.New(config.Agents[i].Id, config.Agents[i].ApiURL, nil)
 		if err != nil {
-			return nil, fmt.Errorf("cluster: failed to create agent: %w", err)
+			return nil, fmt.Errorf("cluster: failed to create api client: %w", err)
 		}
 		agents[i] = agent
 		errMap[agent] = errorTrack{}
+
+		// We check if the agent has already been created.
+		if _, err := agent.Status(); err == nil {
```

Comment:
Do we want to have a dedicated API for this which just returns the id or something? Totally fine with this too. Just thinking out loud.

Reply:
That went through my mind. I was also thinking of having the `Create()` method return a special `ErrExists` error than could be ignored in this case (to avoid making two calls essentially) but in the end didn't settle on it and went for this.
If the resource exists we can return the status of it so I thought this was the closest thing to an `Exists()` method since it's essentially a simple `GET /loadagent/${id}`.

Reply:
Hmm .. okay. Perhaps in future, we can wrap this into an `Exists` method to make things more readable. Fine as is for now.